### ISSUE TITLE
Unnecessary `tabs` permissions

### ISSUE THREAD

**Issue by gorhill** [0x48piraj/fadblock#134]
In your [FAQ](https://github.com/0x48piraj/fadblock#faq), you state:

> Chrome extension says "Read your browsing history"?
> 
> Prime example of Google being nonsensical, again.

Google is not being "nonsensical": asking for the `tabs` permission in the manifest gives an extension the ability to access the URL of all the webpages visited by the user.

In the [Chrome Web Store](https://chromewebstore.google.com/detail/fadblock-friendly-adblock/mdadjjfmjhfcibgfhfjbaiiljpllkbfc) you state:

> This is the permission required for an extension to work with the browser's tabs

The `tabs` API is always accessible, you do not need to declare it in the manifest to be able to access it.

In your manifest, you already declare these host permissions:

>     "https://www.youtube.com/*",
>     "https://*.youtube.com/*",

These will automatically grant you the ability to see the `tab.url` property for webpages which match your declared host permissions, see <https://developer.chrome.com/docs/extensions/reference/tabs/#perms>:

> [Host permissions](https://developer.chrome.com/docs/extensions/mv3/match_patterns/) allow an extension to read and query a matching tab's four sensitive `tabs.Tab` properties. They can also interact directly with the matching tabs using methods such as [tabs.captureVisibleTab()](https://developer.chrome.com/docs/extensions/reference/tabs/#method-captureVisibleTab), [tabs.executeScript()](https://developer.chrome.com/docs/extensions/reference/tabs/#method-executeScript), [tabs.insertCSS()](https://developer.chrome.com/docs/extensions/reference/tabs/#method-insertCSS), and [tabs.removeCSS()](https://developer.chrome.com/docs/extensions/reference/tabs/#method-removeCSS).

Same applies to Firefox, see <https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs>.

---

The `tabs` permission give your extension access to sensitive information which is unnecessary to fulfill its purpose.

Should you ever decide to sell your extension, not having an unnecessary `tabs` permission will ensure users of your extension are not exposed to exfiltration of sensitive information (beyond visited webpages on Youtube) if ever the new owner is unethical.

**Comment by 0x48piraj** [comment]
It was included for backward compatibility; in certain Chromium browsers, the `tabs` permission is necessary.

Google's explanation seems unnecessarily permissive ("nonsensical"), as you pointed out; `tabs` merely allows developers to access URLs, and nothing more. Firefox's description provides a more accurate portrayal of the `tabs` permission.

But thank you for putting this into a cogent issue, I will include this into the FAQ and will see if I can remove the tabs permissions without breaking support.

**Comment by gorhill** [comment]
> certain Chromium browsers, the `tabs` permission is necessary

Can you be specific? Which ones?

> Google's explanation seems unnecessarily `permissive` ("nonsensical"), as you pointed out

I don't understand what that sentence mean. I pointed out I agree with Google's permission warning, and that I disagree with you that it is "nonsensical" -- I was quoting you.

> merely allows developers to access URLs, and nothing more

"Nothing more" except the ability to exfiltrate browsing history. [Example](https://robertheaton.com/2018/07/02/stylish-browser-extension-steals-your-internet-history/):

> Stylish [...] records every single website that I and its 2 million other users visit

**Comment by 0x48piraj** [comment]
I can't recall at the moment but on some versions of Vivaldi/Brave, it didn't work unless `tabs` was defined. I will have to write a test suite to know this for sure.

I meant in Chrome the permission warning states "Read your browsing history" which is a bit permissive, Firefox says "Access browser tabs" which seems more appropriate. Why?

Because, technically, this permission allows observing the history via `chrome.tabs.onUpdated`, but not read the previously existing history, so the phrasing is not entirely correct, strictly speaking.

 That's what I meant. Sorry for not being clear.

But I understand your concern (and that was a fun read, thank you for the resource!), I can certainly eliminate the `tabs` permission from both the Chrome and Firefox versions. As of version 2.6, the `tabs` permission is still present but in the next one, I'll consider removing it, but wouldn't that lead to FadBlock being disabled on every device due to the change of permissions?

**Comment by 0x48piraj** [comment]
Seems like you were onto something... Do all Chrome extensions that are sold end up being misused? It's heartbreaking.

I handed off the extension, it seems the extension traded hands a couple of times this past month as the _current owner is not the same to whom I handed off the extension to_. The new owner is indeed malicious, but fortunately, the updated codebase asked for new permissions, **the extension got flagged and this security issue was never exploited**. 

As of now, I have mirrored the extension under a new name and removed the `tabs` permission from the jump.
